📚 Add developer release instructions (#191)
aiidateam/aiida-pseudo: docs/source/dev.md

* Minor release: `v1.7.1` to `v1.8.0`, bug fixes and new features that **maintain** backwards compatibility.
* Major release: `v1.7.1` to `v2.0.0`, bug fixes and new features that **break** backwards compatibility.

In the steps below, we'll assume the version to be released is `v1.7.2`.

### Steps

To make a new release of the `aiida-pseudo` plugin package, first create a release branch based on the latest `main` branch:

```console
git fetch --all --prune
git checkout origin/main -b release/1.7.2
```

Next, update the source code `__version__` in the `src/aiida_pseudo/__init__.py` file by hand.
It should look something like this:

```
"""AiiDA plugin that simplifies working with pseudo potentials."""
__version__ = '1.7.2'
```

Then, run the `update_changelog.py` script to update the `CHANGELOG.md` with all the changes made since the last release:

```console
python .github/workflows/update_changelog.py
```

This will automatically add:

1. The header for the new release and the sub-headers in which the commits should be sorted.
2. The list of commits since the previous release, with links to them.

Sort the commit items into the correct subsection of the change log.

> Ideally, the main changes or new features are also described at the top of the change log message, providing code snippets where it's useful.

Reviewer: This passive voice is a bit confusing. Do you mean one SHOULD do this, as in "Ideally, describe the main changes or new features..."?

Author: Yes, ideally, but time is always a factor 😉 . Let me adapt!

> Once you've prepared the release branch locally, commit  and push it to Github:

Reviewer: ```suggestion
Once you've prepared the release branch locally, commit and push it to Github:
```

    git commit -am '🚀 Release `v1.7.2`'
    git push -u origin release/1.7.2

> Now that the release branch is ready, merge it into `main` via a pull request.
> Make sure the remote `release/1.7.2` branch is up to date by pushing the local changes, then go to Github and create a pull request to the `main` branch of the official repository.

Reviewer: ```suggestion
Now that the release branch is ready, push the local changes to the remote `release/1.7.2` branch, then go to Github and create a pull request to the `main` branch of the official repository.
```

> After the pull request has been approved, merge the PR using the "Squash and Merge", and make sure the commit is simply named e.g. "🚀 Release `v1.7.2`".

Reviewer: ```suggestion
After the pull request has been approved, merge the PR using the **"Squash and Merge"**, and make sure the commit is simply named e.g. "🚀 Release `v1.7.2`".
```

Reviewer: I've missed this at least once, so maybe good to emphasize 😅
